### PR TITLE
Kafka service

### PR DESCRIPTION
**Pull request by m4ce**
Hi,

Since I needed to run Kafka in our environment, I thought I'd contribute back and prepare this pull request.

Let me know if I should make any changes.

Cheers,
Matteo

### CODE REVIEW

**Comment by cloudnautique** [comment]
Awesome! I'm excited to try it out.

File: kafka/containers/0.1.0/kafka-conf/templates/zookeeper_connect.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{{range ls "/services/zookeeper/containers"}}{{ $containerInstance := getv (printf "/services/zookeeper/containers/%s" .)}}
```
Comment (cloudnautique) [inline kafka/containers/0.1.0/kafka-conf/templates/zookeeper_connect.yml:1]: The file name needs to be zookeeper_connect.tmpl

File: kafka/containers/0.1.0/kafka/run.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# wait for kafka to start up
+while ! netstat --listen --numeric --tcp | awk '{ print $4 }' | egrep -q ":$KAFKA_CONFIG_PORT$"; do
+  # make sure we do not wait forever though
+  kill -0 $pid &>/dev/null || { echo "Kafka shut down unexpectedly"; exit 1; }
+  echo "Waiting for kafka to bind to TCP $KAFKA_CONFIG_PORT"
+  sleep 1
+done
+
+if [ -n "$KAFKA_CREATE_TOPICS" ]; then
```
Comment (cloudnautique) [inline kafka/containers/0.1.0/kafka/run.sh:56]: Can this logic be in its own sidekick? Possibly define the topics from metadata on the service? 

```
rancher-compose.yml:
  kafka:
    scale: 3
    metadata:
        topics:
           - name:
             partition:
             replica:
```

**Comment by cloudnautique** [comment]
I seem to have run into some networking errors... It looks like kafka ends up bound to the 172.17.x IP and not the Rancher IP. I get WARN Fetching topic metadata with correlation id 0 for topics when trying to use it. It sounds like we want to advertise the host through it RancherIP(quick google came up with https://issues.apache.org/jira/browse/KAFKA-1092).. I'm not very familiar with Kafka, I was going through the quick start guide on apache site and can't get messages.

**Comment by ruimarinho** [comment]
With Kafka 0.10 out with [rack awareness](https://cwiki.apache.org/confluence/display/KAFKA/KIP-36+Rack+aware+replica+assignment), there is a lot of potential for a proper integration with Rancher. Is anyone working on this PR?

**Comment by cjellick** [comment]
Very stale. Closing

**Pull request closed without merging**